src/lib/multikey_dict.py

```diff
@@ -0,0 +1,109 @@
+"""Class for multikey dicts."""
+
+from typing import Hashable, Union
+
+
+class MultiKeyDict:
+    """Utility class for supporting data with multiple lookup keys.
```

Comment:
Need more strict wording here.

MultiKeyDict is a class that handles mapping multiple keys to the same value.

Values are keyed by a tuple of keys, of which any can be used to look up the value.
The size of the tuple is user specified, fixed, and must be known at instantiation time. Adding values to the dict
requires a complete tuple of keys, but lookups can be done with any key in the tuple.

Reply:
Updated docs with strict wording, added more information about limitations on removal/modification of multikeys.